From eebb1568f9f8c9a5514dfffcd7d5eaf98d80b402 Mon Sep 17 00:00:00 2001
From: Soumya Koduri <skoduri@redhat.com>
Date: Thu, 9 Jul 2015 15:46:55 +0530
Subject: [PATCH 11/20] FSAL_GLUSTER: Include 'O_EXCL' flag while trying to create a file

NFSv4 OPEN (especially EXCLUSIVE create) expects ERR_EXIST error
in case if the file already exists.

Change-Id: I0b0ed48565d100983cd970f5c28da2b8c7e9aae0
Signed-off-by: Soumya Koduri <skoduri@redhat.com>
---
 src/FSAL/FSAL_GLUSTER/handle.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/FSAL/FSAL_GLUSTER/handle.c b/src/FSAL/FSAL_GLUSTER/handle.c
index a750b3d..a9c6db0 100644
--- a/src/FSAL/FSAL_GLUSTER/handle.c
+++ b/src/FSAL/FSAL_GLUSTER/handle.c
@@ -244,7 +244,7 @@ static fsal_status_t create(struct fsal_obj_handle *dir_hdl,
 	/* FIXME: what else from attrib should we use? */
 	glhandle =
 	    glfs_h_creat(glfs_export->gl_fs, parenthandle->glhandle, name,
-			 O_CREAT, fsal2unix_mode(attrib->mode), &sb);
+			 O_CREAT | O_EXCL, fsal2unix_mode(attrib->mode), &sb);
 
 	rc = setglustercreds(glfs_export, NULL, NULL, 0, NULL);
 	if (rc != 0) {
-- 
1.7.1

